### PR TITLE
Jonathan/caseflow 4473 emo assign to rpo

### PR DESCRIPTION
**Pull request by jtsangVA**
Resolves https://vajira.max.gov/browse/CASEFLOW-4473

### Description
Implemented a new task action and subsequent modal for EMO users to successfully assign education predocket appeals to RPO

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] "Assign to Regional Processing Office" task action is created
- [ ] "Assign to RPO" modal is created
- [ ] Modal allows for assignment to any of the three available RPOs (Buffalo, Central Processing Office, Muskogee)
- [ ] Instructions are optional per designs
- [ ] EMO can successfully assign a predocket appeal to RPO
- [ ] Upon success, show the appropriate success banner (most likely a Success Alert in the Queue view however awaiting design feedback)
- [ ] Upon fail, show a failure (most likely a Slim Error Alert in the modal)
- [ ] After assignment the EMO task has a status of "on hold"
- [ ] The RPO task should be created and have a status of "assigned"

### Testing Plan
1. As a BVA Intake user, submit an Appeal with an Education issue and require pre-docketing.
2. Switch to an EMO user.
3. Navigate to the EMO organization queue: http://localhost:3000/organizations/edu-emo?tab=education_emo_unassigned&page=1
4. Ensure that the appeal you just created appears in the Unassigned tab.
5. Click on the appeal to go to the Case Details page.
6. Click on the Actions dropdown list and select "Assign to Regional Processing Office"
7. In the Modal, select a RPO and click the Submit button.
8. Verify that the success banner matches the RPO that you selected.
9. Click on the EMO Assigned Tab and verify that the appeal has moved over.
10. Switch users to an EMO RPO user.
11. Navigate to /queue and click the Switch Views button. Select the RPO that you assigned the appeal to in the previous modal.
12. Verify that the appeal appears in the RPO Assigned tab.
13. Open the rails console and inspect the Task Tree. Ensure that the appeal has the correct Tasks, Statuses and Assignees. 
image

- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 image
 local
image

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### CODE REVIEW

**Review by ThorntonMatthew — COMMENTED** [review]
Yours | Design's
---|---
present|design


The only difference between your implementation and the designs is the success message shown after assigning a task to an RPO. I'm not sure how exact it has to be, but I wanted to make note of it.

It looks like the success message is being partially determined in [assignTaskSuccessMessage](https://github.com/department-of-veterans-affairs/caseflow/blob/a9229a948cc0bd17c6e31b04577e4a036e4b4f17/client/app/queue/AssignToView.jsx#L90) with an additional ternary statement below in the [return](https://github.com/department-of-veterans-affairs/caseflow/blob/a9229a948cc0bd17c6e31b04577e4a036e4b4f17/client/app/queue/AssignToView.jsx#L105).

File: app/repositories/task_action_repository.rb [hunk 1]
```diff
@@ -586,6 +586,18 @@ def emo_return_to_board_intake(*)
       }
     end
 
+    def emo_assign_to_regional_processing_office_data(*)
+      {
+        options: organizations_to_options(EduRegionalProcessingOffice.all),
+        modal_title: COPY::EMO_ASSIGN_TO_REGIONAL_PROCESSING_OFFICE_MODAL_TITLE,
+        modal_body: COPY::PRE_DOCKET_MODAL_BODY,
+        modal_selector_placeholder: COPY::EMO_REGIONAL_PROCESSING_OFFICE_SELECTOR_PLACEHOLDER,
```
Comment (ThorntonMatthew) [inline app/repositories/task_action_repository.rb:594]: Do you think this should be `EDU_REGIONAL_PROCESSING_OFFICE_SELECTOR_PLACEHOLDER` to match the `EduRegionalProcessingOffice` class name?

File: client/app/queue/AssignToView.jsx [hunk 2]
```diff
@@ -54,6 +54,12 @@ class AssignToView extends React.Component {
   componentDidMount = () => this.props.resetSuccessMessages();
 
   validateForm = () => {
+    const actionData = taskActionData(this.props);
+
+    if (actionData.body_optional) {
```
Comment (ThorntonMatthew) [inline client/app/queue/AssignToView.jsx:59]: This is so slick!!

File: app/repositories/task_action_repository.rb (hunk 1 above)
Comment (jtsangVA) [reply app/repositories/task_action_repository.rb:594]: Good catch, this was a typo.

**Review by ThorntonMatthew — APPROVED** [review]
lgtm!!

**Pull request merged**